hackathon_site/event/api_views.py

```diff
@@ -66,3 +81,57 @@ def post(self, request, *args, **kwargs):
         response_serializer = TeamSerializer(profile.team)
         response_data = response_serializer.data
         return Response(data=response_data, status=status.HTTP_201_CREATED,)
+
+
+class JoinTeamView(generics.GenericAPIView, mixins.RetrieveModelMixin):
+    permission_classes = [UserHasProfile]
+    serializer_class = TeamSerializer
+
+    def clean(self):
+        if not is_registration_open():
+            raise forms.ValidationError(
+                _("You cannot change teams after registration has closed."),
+                code="registration_closed",
+            )
+
+        return super().clean()
+
+    @transaction.atomic
+    def post(self, request, team_code, *args, **kwargs):
+
+        profile = request.user.profile
+        current_team = profile.team
+
+        try:
+            team = Team.objects.get(team_code=team_code)
+        except:
+            raise ValidationError(
+                {"detail": "Team does not exist!"}, code=status.HTTP_400_BAD_REQUEST
+            )
+
+        if not team.profiles.exists():
+            raise ValidationError(
+                {"detail": "Cannot join a fully empty team!"},
+                code=status.HTTP_400_BAD_REQUEST,
+            )
+        if team.profiles.count() >= Team.MAX_MEMBERS:
+            raise ValidationError({"detail": "Team is full"})
+
+        # Raise 400 if team has active orders
+        active_orders = OrderItem.objects.filter(
+            ~Q(order__status="Cancelled"), Q(order__team=team),
+        )
+
+        if active_orders.exists():
+            raise ValidationError(
+                {"detail": "Cannot join a team with already processed orders"},
+                code=status.HTTP_400_BAD_REQUEST,
+            )
+
+        print(type(profile.team))
+        profile.team = team
```

Comment:
You haven't handled cleaning up their old team at all

Reply:
now, members cannot join another team if current team has active orders, as per expectation/description of JIRA issue

if old team only had one member (themselves), the team is deleted.